catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/reader/GetRecordsMessageBodyReader.java

```diff
@@ -142,10 +147,15 @@ public CswRecordCollection readFrom(
           new XppReader(
               new InputStreamReader(inStream, StandardCharsets.UTF_8),
               XmlPullParserFactory.newInstance().newPullParser());
+
       cswRecords = (CswRecordCollection) xstream.unmarshal(reader, null, argumentHolder);
+    } catch (CannotResolveClassException cre) {
+      // This exception can happen if the XppReader encounters unexpected (but correct) namespaces
+      // prefixes
+      // In this case, attempt to use the StaxReader which is namespace aware.
+      cswRecords = unmarshalWithStaxReader(originalInputStream);
```

Comment:
I am guessing there is a reason why we wouldn't just do it this way all the time?  Is this basically the old way for when it works (registry) and then deal with it here when it doesn't?  I like the less invasive approach, considering our circumstances, just wanted to make sure I understand it.

Reply:
Yeah, it feels like if there is a way that handles namespaces correctly, then we should use that all the time.

Reply:
The StaxReader processes namespace prefix mappings correctly during the unmarshalling, however it doesn't pass the namespace declarations/prefixes to the underlying converter.  This addresses the primary issue we have seen where certain CSW responses use alternative namespace prefix mappings. The CSW Record Converter doesn't care about the namespace prefixes being there (it parses the fields/attributes using the xstream API).  However, the RegistryTransformer utilizes JAXB and relies on well-formed XML including namesapces declarations.  I haven't found a way to tell the StaxReader to preserve the namespace information.   The XppReader does NOT process namespace declarations correctly, but it DOES pass the full XML document to the underlying Converters.  I haven't found one xstream XML reader that works in both cases.